Sources/GRPCCore/GRPCClient.swift

```diff
@@ -361,14 +406,50 @@ public final class GRPCClient: Sendable {
     var options = options
     options.formUnion(with: methodConfig)
 
+    let applicableInterceptors = self.state.withLock {
```

Comment:
Rather than doing all of this inline can we do it in a method on the `StateMachine`?

Also, it'd be good if we can combine this with `checkExecutable` above to avoid acquiring the lock twice per RPC.